apps/web/app/(app)/environments/[environmentId]/lib/project.ts

```diff
@@ -0,0 +1,51 @@
+import { validateInputs } from "@/lib/utils/validate";
+import { Prisma } from "@prisma/client";
+import { cache as reactCache } from "react";
+import { prisma } from "@formbricks/database";
+import { ZString } from "@formbricks/types/common";
+import { DatabaseError } from "@formbricks/types/errors";
+import { TMembership, ZMembership } from "@formbricks/types/memberships";
+
+export const getUserProjects = reactCache(
+  async (userId: string, orgMembership: TMembership): Promise<{ id: string; name: string }[]> => {
+    validateInputs([userId, ZString], [orgMembership, ZMembership]);
+
+    let projectWhereClause: Prisma.ProjectWhereInput = {};
+
+    if (orgMembership.role === "member") {
+      projectWhereClause = {
+        projectTeams: {
+          some: {
+            team: {
```

Comment:
Does this query work when team functionality is not used?
If they have an enterprise license and don't have access to the advanced access control feature they can assign users the `member` role, but don't have teams and thus the members won't see any projects.